### PR TITLE
split action package into action and logging packages; add TestIsatty

### PR DESCRIPTION
**Pull request by anmyachev**

### CODE REVIEW

File: pkg/logging/logging.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ import (
 )
 
 var (
-	logWriter  *io.PipeWriter
+	LogWriter  *io.PipeWriter
 	logExt     string = ".log"
 	timeLayout string = "2006-01-02-15-04-05"
-	repeatLogs bool   = false
+	RepeatLogs bool   = false
```
Comment (vnlitvinov) [inline pkg/logging/logging.go]: I don't like making this variable public, if you need to access it make a getter

File: pkg/logging/logging.go [hunk 2]
```diff
@@ -14,10 +14,10 @@ import (
 )
 
 var (
-	logWriter  *io.PipeWriter
+	LogWriter  *io.PipeWriter
```
Comment (vnlitvinov) [inline pkg/logging/logging.go]: make a getter probably

File: pkg/logging/logging_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+func TestIsatty(t *testing.T) {
+	if tty := isatty(os.Stdout); tty != false {
+		t.Errorf("os.Stdout is not terminal; isatty should return false")
+	}
+
+	_, writer := io.Pipe()
+	if tty := isatty(writer); tty != false {
+		t.Errorf("PipeWriter is not terminal; isatty should return false")
+	}
+	defer writer.Close()
```
Comment (vnlitvinov) [inline pkg/logging/logging_test.go]: this should be done right after creating the writer; also you should probably check error result of `io.Pipe()` if there is any

Comment (anmyachev) [reply pkg/logging/logging_test.go]: `io.Pipe()` doesn't return error

Comment (anmyachev) [reply pkg/logging/logging_test.go]: removed refer

File: pkg/logging/logging.go (hunk 1 above)
Comment (anmyachev) [reply pkg/logging/logging.go]: getters were added

**Comment by anmyachev** [comment]
> Why split?

`action` package functionality is well divided into logging and execution of actions

File: pkg/action/action.go [hunk 4]
```diff
@@ -27,11 +29,12 @@ func RunLoggedCmdDirOutput(logfilePrefix string, workDir string, output io.Write
...
 	var cmdLog io.Writer
 
-	logname, cmdLogfile, err := makeLogWriter(logfilePrefix)
+	logname, cmdLogfile, err := logging.MakeLogWriter(logfilePrefix)
 	if err != nil {
 		log.Warnf("RunLoggedCmdDirOutput: cannot redirect log to file: %s, using default log instead", err)
 
-		cmdLog = logWriter
+		logWriter := logging.GetLogWriter()
+		cmdLog = &logWriter
```
Comment (semelianova) [inline pkg/action/action.go]: Why do you create an additional variable `logWriter` ? Why not initialize `cmdLog` directly?

Comment (anmyachev) [reply pkg/action/action.go]: I think you speak about this version: `cmdLog = &logging.GetLogWriter()`
In the case Go returns next error: `pkg\action\action.go:36:12: cannot take the address of logging.GetLogWriter()`

Comment (semelianova) [reply pkg/action/action.go]: got it thanks

File: pkg/logging/logging.go [hunk 5]
```diff
@@ -20,6 +20,16 @@ var (
 	repeatLogs bool   = false
 )
 
+// GetLogWriter gives access to `logWriter` without the possibility of changing it
+func GetLogWriter() io.PipeWriter {
```
Comment (vnlitvinov) [inline pkg/logging/logging.go]: return type should be generic `io.Writer`, of this package don't have to know that it's a pipe
```suggestion
func GetLogWriter() io.Writer {
```

Comment (anmyachev) [reply pkg/logging/logging.go]: fixed

File: pkg/logging/logging.go [hunk 6]
```diff
@@ -21,8 +21,11 @@ var (
 )
 
 // GetLogWriter gives access to `logWriter` without the possibility of changing it
-func GetLogWriter() io.PipeWriter {
-	return *logWriter
+func GetLogWriter() io.Writer {
+	// explicit copy
+	logger := *logWriter
```
Comment (vnlitvinov) [inline pkg/logging/logging.go]: shouldn't be needed as you're returning an interface... but that won't hurt much, so let's keep it

**Pull request merged**